### PR TITLE
closes RFC #21: streamlining the boilerplate

### LINKED ISSUE #15: creating target directory in parent of root
Something about the latest version of the gypfile is causing the creation of a `target` directory in the parent of the project root.

### LINKED ISSUE #18: can we kill binding.cc?
Both @eddyb and @mmun have brought up the fact that binding.cc is boring and an eyesore. Is it possible to implement it in pure Rust? How hard is it to implement what `NODE_MODULE` is doing?

### PR DESCRIPTION
**Pull request by dherman**
NOTE: this WIP patch depends on rustbridge/neon#35
- also closes #15 and closes #18
- `neon new` creates new project structure
  - `native/` subdirectory
  - no more neon-bridge
- adds `neon build`, which builds with cargo and generates index.node
- CLI is all async using promises and async/await

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,28 +14,27 @@
...
     "git-config": "0.0.7",
     "handlebars": "^4.0.3",
     "in-publish": "^2.0.0",
     "inquirer": "^0.11.0",
     "minimist": "^1.2.0",
-    "mkdirp": "^0.5.1",
-    "nan": "2.0.8",
-    "neon-bridge": "^0.1.3",
-    "node-gyp": "3.0.3",
+    "nexpect": "^0.5.0",
```
Comment (mmun) [inline package.json]: Can this stay a devDep?

File: package.json [hunk 2]
```diff
@@ -14,28 +14,27 @@
...
+    "rsvp": "^3.1.0",
     "semver": "^5.1.0",
+    "toml": "^2.3.0",
```
Comment (mmun) [inline package.json]: Ditto

**Comment by eddyb** [comment]
This doesn't seem to be removing the template files which are no longer used.

**Comment by dherman** [comment]
@eddyb Thanks, fixed.

**Comment by dherman** [comment]
OK just got `neon new` working and push -f'ed. Let's go Travis...

File: src/ops/neon_build.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  return spawn(command, args, { cwd: 'native', stdio: 'inherit', env: env });
+}
+
+async function main(name, configuration) {
+  let pp = process.platform;
+  let dylib = path.resolve('native', 'target', configuration, LIB_PREFIX[pp] + name + LIB_SUFFIX[pp]);
+  let index = path.resolve('native', 'index.node');
+
+  console.log(style.info("generating native/index.node"));
```
Comment (watilde) [inline src/ops/neon_build.js:53]: Tiny tweak for Windows

``` diff
+ console.log(style.info("generating native" + path.sep + "index.node"));
- console.log(style.info("generating native/index.node"));
```

Comment (dherman) [inline src/ops/neon_build.js:53]: Oops! Sorry I merged too quickly. I'll fix in a sec.

**Pull request merged**